guides/testing.md

```diff
@@ -0,0 +1,14 @@
+# Testing Guidelines
+
+## Use Jest's mock functions instead of Sinon
+
+Jest incorporates most of the features of Sinon with a slimmer API:
+
+* `jest.fn()` can be used in place of `sinon.stub()`.
+* `jest.spyOn(object, method) can be used in place of `sinon.spy(object, method)`.
+* `jest.spyOn(object, method)` can be used in place of `sinon.stub(object, method)` (with the caveat that the method being spied upon will still be called by default).
+* `jest.useFakeTimers()` can be used in place of `sinon.useFakeTimers()` (though note that Jest's "clock" object had fewer features than Sinon's prior to Jest v29.5).
+
+## Don't test private code
```

Comment:
It would be nice to expand more on the reasoning behind this.
e.g.
* Explain how this ties in with a broader BDD-like testing philosophy,
* Tests are more useful for preventing regressions and helping document intended behavior when they only reflect the public API
* This ensures the tests help make refactoring easier, rather than getting in the way

This is a great start though! I can follow up to add more detail later.

Reply:
Sounds good!